src/benchmarks/micro/corefx/System.Buffers/BinaryReadAndWriteTests.cs

```diff
@@ -0,0 +1,300 @@
+        }
+
+        [Benchmark]
+        public TestStructExplicit ReadStructFieldByFieldUsingBitConverterLE()
+        {
+            byte[] arrayLE = _arrayLE;
+
+            var readStruct = new TestStructExplicit
+            {
+                S0 = BitConverter.ToInt16(arrayLE, 0),
+                I0 = BitConverter.ToInt32(arrayLE, 2),
+                L0 = BitConverter.ToInt64(arrayLE, 6),
+                US0 = BitConverter.ToUInt16(arrayLE, 14),
+                UI0 = BitConverter.ToUInt32(arrayLE, 16),
+                UL0 = BitConverter.ToUInt64(arrayLE, 20),
+                S1 = BitConverter.ToInt16(arrayLE, 28),
+                I1 = BitConverter.ToInt32(arrayLE, 30),
+                L1 = BitConverter.ToInt64(arrayLE, 34),
+                US1 = BitConverter.ToUInt16(arrayLE, 42),
+                UI1 = BitConverter.ToUInt32(arrayLE, 44),
+                UL1 = BitConverter.ToUInt64(arrayLE, 48),
+            };
+
+            return readStruct;
+        }
+
+        [Benchmark]
+        public TestStructExplicit ReadStructFieldByFieldUsingBitConverterBE()
+        {
+            byte[] arrayBE = _arrayBE;
+
+            var readStruct = new TestStructExplicit
+            {
+                S0 = BitConverter.ToInt16(arrayBE, 0),
+                I0 = BitConverter.ToInt32(arrayBE, 2),
+                L0 = BitConverter.ToInt64(arrayBE, 6),
+                US0 = BitConverter.ToUInt16(arrayBE, 14),
+                UI0 = BitConverter.ToUInt32(arrayBE, 16),
+                UL0 = BitConverter.ToUInt64(arrayBE, 20),
+                S1 = BitConverter.ToInt16(arrayBE, 28),
+                I1 = BitConverter.ToInt32(arrayBE, 30),
+                L1 = BitConverter.ToInt64(arrayBE, 34),
+                US1 = BitConverter.ToUInt16(arrayBE, 42),
+                UI1 = BitConverter.ToUInt32(arrayBE, 44),
+                UL1 = BitConverter.ToUInt64(arrayBE, 48),
+            };
+
+            if (BitConverter.IsLittleEndian)
+            {
+                readStruct.S0 = ReverseEndianness(readStruct.S0);
+                readStruct.I0 = ReverseEndianness(readStruct.I0);
+                readStruct.L0 = ReverseEndianness(readStruct.L0);
+                readStruct.US0 = ReverseEndianness(readStruct.US0);
+                readStruct.UI0 = ReverseEndianness(readStruct.UI0);
+                readStruct.UL0 = ReverseEndianness(readStruct.UL0);
+                readStruct.S1 = ReverseEndianness(readStruct.S1);
+                readStruct.I1 = ReverseEndianness(readStruct.I1);
+                readStruct.L1 = ReverseEndianness(readStruct.L1);
+                readStruct.US1 = ReverseEndianness(readStruct.US1);
+                readStruct.UI1 = ReverseEndianness(readStruct.UI1);
+                readStruct.UL1 = ReverseEndianness(readStruct.UL1);
+            }
+
+            return readStruct;
+        }
+
+        [Benchmark]
+        public int[] MeasureReverseEndianness()
+        {
+            var local = _oneThousandIntegers;
```

Comment:
Nit: var => whatever type it is... same goes for any other occurrences where the type isn't obvious from the RHS